readers/api/transport.go

```
@@ -210,25 +210,35 @@ func encodeError(_ context.Context, err error, w http.ResponseWriter) {
 	}
 }
 
-func authorize(r *http.Request, chanID string) error {
+func authorize(ctx context.Context, r *http.Request, chanID string) (err error) {
 	token := r.Header.Get("Authorization")
 	if token == "" {
 		return errUnauthorizedAccess
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
-	defer cancel()
+	if _, err := auth.CanAccessByKey(ctx, &mainflux.AccessByKeyReq{Token: token, ChanID: chanID}); err == nil {
+		return nil
+	}
```

I'd prefer if we put authorization type somewhere ( `Authorization` header prefix or in a separate header or something like that) so that we know which method to use, rather than implicitly fallback to user if thing authorization fails.

---

That sounds ok to me, I propose the following 
`Authorization: <type> <value>`
So for thing key - `Authorization: Thing <value>`
And for user token - `Authorization: Bearer <value>`
What do you say?
Seems like the right way - https://stackoverflow.com/a/43164958/2873439
@drasko ?

---

Makes sense. The problem is that the prefix will break compatibility with the current authorization header. We can support both (`Things` prefix and no prefix for Thing authorization, `Bearer` for user) to give some space for migration and make prefix required in the next release. Despite the most popular schemas being `Basic` and `Bearer`, I don't see anything wrong with the custom `Thing` schema for thing authorization.